kohvis/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+#[macro_use]
+extern crate gdnative;
+extern crate kohonen;
+
+mod colors;
+mod kohonen_gd;
+mod mapping_gd;
+mod tabs_gd;
+pub use kohonen_gd::Kohonen;
+pub use mapping_gd::Mapping;
+pub use tabs_gd::Tabs;
+
+use gdnative::{Instance, NodePath};
+
+fn init(handle: gdnative::init::InitHandle) {
+    handle.add_class::<Kohonen>();
+    handle.add_class::<Mapping>();
+    handle.add_class::<Tabs>();
+}
+
+trait KohonenUser2D {
+    fn kohonen_path(&self) -> &str;
+
+    fn with_kohonen<F>(mut owner: gdnative::Control, path: &str, mut fun: F)
+    where
+        F: FnMut(&mut gdnative::Control, &Kohonen),
```

Comment:
As the closure is called only once, maybe `FnOnce` would be a better type? And since `Control: Copy` and only 8 bytes large, it is probably more efficient to pass it by value?

Reply:
(`util::with_kohonen` does seem to use `FnMut` as well.)

Reply:
Thanks!